dnfdaemon-server/main.cpp

```diff
@@ -22,12 +22,25 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 #include "session_manager.hpp"
 
 #include <sdbus-c++/sdbus-c++.h>
+#include <signal.h>
 
 #include <iostream>
 
 static std::unique_ptr<SessionManager> session_manager = nullptr;
 
+void sigint_handler([[maybe_unused]] int signum) {
+    if (session_manager != nullptr) {
```

Comment:
As an addendum to the other note on synchronization, we have another possible race here, albeit it would seem less practically important.

You create the `session_manager` as the first thing in `main()`, but if right in the middle of writing to the static `session_manager` variable this signal handler is invoked, it may have an incomplete value set in the `unique_ptr`. Given the unlikelihood and the fact it would probably cause a segfault of a program that has already been terminated we could just leave this.

Otherwise this is what we'd use `std::atomic` for, but I'm not sure how it works for a `unique_ptr` exactly. From https://en.cppreference.com/w/cpp/atomic/atomic it seems it's not supported. You could, though, store the owner of the pointer inside `main()` and use either a raw pointer or `std::weak_ptr` inside the atomic. In fact, the cleanest would probably be `SessionManager` in an on-the-stack variable in `main()` and a `static std::atomic<SessionManager *>` global variable pointing to it.